### ISSUE TITLE
Should casting __intcap_t to void * be allowed without a __cheri_fromcap?

### ISSUE THREAD

**Issue by brooksdavis** [CTSRD-CHERI/llvm-project#315]
This currently compiles without warnings:
```C
void *
afunc(__intcap_t foo)
{
        return (void *)foo;
}
```
If `foo` is a valid capability then the result may not be as expected.  I found this when the compiler missed a `copyin((void *)an_intcap_t, ...)` that needed special handling

**Comment by jrtc27** [comment]
This warns these days with:
```
<source>:4:16: warning: the following conversion will result in a CToPtr operation; the behaviour of CToPtr can  be confusing since using CToPtr on an untagged capability will give 0 instead of the  integer value and should therefore be explicitly annotated [-Wcheri-pointer-conversion]
        return (void *)foo;
               ^~~~~~~~~~~
<source>:4:16: note: if you really intended to use CToPtr use __builtin_cheri_cap_to_pointer() or a __cheri_fromcap cast to silence this warning; to get the virtual address use __builtin_cheri_address_get() or a __cheri_addr cast; to get the capability offset use __builtin_cheri_offset_get() or a __cheri_offset cast
        return (void *)foo;
               ^~~~~~~~~~~
1 warning generated.
```
